Userland/Utilities/comm.cpp

```diff
@@ -93,35 +103,41 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     auto cmp = [&](String const& str1, String const& str2) {
         if (case_insensitive)
             return strcasecmp(str1.characters(), str2.characters());
-        else
-            return strcmp(str1.characters(), str2.characters());
-    };
-
-    auto process_remaining = [](String const& fmt, auto& file, int& count, bool print) {
-        while (file->can_read_line()) {
-            ++count;
-            auto line = file->read_line();
-            if (print)
-                outln(fmt, line);
-        }
+        return strcmp(str1.characters(), str2.characters());
     };
 
     bool read_file1 { true };
     bool read_file2 { true };
     int col1_count { 0 };
     int col2_count { 0 };
     int col3_count { 0 };
-    String file1_line, file2_line;
+    String file1_line;
+    String file2_line;
+    Array<u8, PAGE_SIZE> buffer;
+
+    auto should_continue_comparing_files = [&]() {
+        if (read_file1) {
+            auto can_read_file1_line = file1->can_read_line();
+            if (can_read_file1_line.is_error() || !can_read_file1_line.value())
+                return false;
+        }
+        if (read_file2) {
+            auto can_read_file2_line = file2->can_read_line();
+            if (can_read_file2_line.is_error() || !can_read_file2_line.value())
```

Comment:
`return !can_read_file?_line.is_error() && can_read_file?_line.value()` looks better IMO.

Reply:
We only want to return false if it's wrong - if it's not, we carry on to the next part. (Otherwise, if `read_file1` and `read_file2` were both true, we'd only check file 1.)